src/observers/link_prefetch_observer.js

```diff
@@ -100,14 +100,11 @@ export class LinkPrefetchObserver {
 
     request.headers["Sec-Purpose"] = "prefetch"
 
-    if (link.dataset.turboFrame && link.dataset.turboFrame !== "_top") {
-      request.headers["Turbo-Frame"] = link.dataset.turboFrame
-    } else if (link.dataset.turboFrame !== "_top") {
-      const turboFrame = link.closest("turbo-frame")
+    const turboFrame = link.closest("turbo-frame")
```

Comment:
Is checking for an ancestor `<turbo-frame>` sufficient here? What if the `<a>` element has a `[data-turbo-frame]` attribute?

Reply:
That's what the `link.dataset.turboFrame` in the line below is doing. I'll update it to use `getAttribute("data-turbo-frame")` so that's hopefully clearer.